pkg/util/cluster/cluster.go

```diff
@@ -193,20 +223,26 @@ func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName str
 		return err
 	}
 
+	diskEncryptionSet, err := c.diskEncryptionSets.Get(ctx, vnetResourceGroup, diskEncryptionSetName)
```

Comment:
I think at this point we should probably rename `vnetResourceGroup` to something more generic as it now hold not only vnet, but key vault as well. I however don't have ideas apart from something stupid like "bring your own resource resource group" :)

Reply:
Also specificly for this line:

You do `c.diskEncryptionSets.Get(...)` here but you only use it to retreive `diskEncryptionSet.ID`. At the same time, few lines below you construct this ID (line 235):

```go
{"/subscriptions/" + c.env.SubscriptionID() + "/resourceGroups/" + vnetResourceGroup + "/providers/Microsoft.Compute/diskEncryptionSets/" + diskEncryptionSetName, rbac.RoleReader},
```